src/Parser/GeoJsonParser.js

```diff
@@ -59,108 +59,65 @@ function readCoordinates(crsIn, crsOut, coordinates, extent, target) {
 // - Coordinates is an array of Coordinate
 // - extent is optional, it's coordinates's extent
 // Multi-* geometry types are merged in one.
+let globalOffset = 0;
 const GeometryToCoordinates = {
     point(crsIn, crsOut, coordsIn, filteringExtent, options) {
         const extent = options.buildExtent ? new Extent(crsOut, Infinity, -Infinity, Infinity, -Infinity) : undefined;
         let coordinates = readCoordinates(crsIn, crsOut, coordsIn, extent);
         if (filteringExtent) {
             coordinates = coordinates.filter(c => filteringExtent.isPointInside(c));
         }
-        return { type: 'point', vertices: coordinates, extent };
+
+        return [coordinates, [{ extent }]];
     },
     polygon(crsIn, crsOut, coordsIn, filteringExtent, options) {
         const extent = options.buildExtent ? new Extent(crsOut, Infinity, -Infinity, Infinity, -Infinity) : undefined;
         // read contour first
         const coordinates = readCoordinates(crsIn, crsOut, coordsIn[0], extent);
         if (filteringExtent && !filteringExtent.isPointInside(coordinates[0])) {
-            return;
+            return [];
         }
-        const contour = {
-            offset: 0,
-            count: coordinates.length,
-        };
+        const indices = [globalOffset, coordinates.length];
```

Comment:
The `globalOffset` bit could be handled at the calling site instead of here. And would allow to remove the global var.

Reply:
Also I think having an array of objects `{ offset: , count: }` would be clearer and less error prone.

Reply:
I changed that, and got back to an object containing offset and count